csharp/src/Ice/ColocatedConnector.cs

```diff
@@ -50,6 +52,8 @@ public Connection Connect(string connectionId)
 
         public override string ToString() =>
             _endpoint.Adapter.Name.Length == 0 ? "unnamed adapter" : _endpoint.Adapter.Name;
+        public override bool Equals(Connector? obj) => ReferenceEquals(this, obj);
```

Comment:
So 2 connectors constructor over the same ColocatedEndpoint are different. Is this as intended?

Reply:
I will update it to compare the endpoints, but we only create one ColocatedEndpoint per adapter and cache it